zenoh-protocol/src/link/unix.rs

```diff
@@ -0,0 +1,921 @@
+//
+// Contributors:
+//   ADLINK zenoh team, <zenoh@adlink-labs.tech>
+//
+use async_std::os::unix::net::{UnixListener, UnixStream};
+use async_std::path::{Path, PathBuf};
+use async_std::prelude::*;
+use async_std::sync::{channel, Arc, Barrier, Mutex, Receiver, RwLock, Sender, Weak};
+use async_std::task;
+use async_trait::async_trait;
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::fmt;
+use std::fs::remove_file;
+use std::net::Shutdown;
+use std::time::Duration;
+use uuid::Uuid;
+
+use super::{Link, LinkTrait, Locator, ManagerTrait};
+use crate::io::{ArcSlice, RBuf};
+use crate::proto::SessionMessage;
+use crate::session::{Action, SessionManagerInner, Transport};
+use zenoh_util::core::{ZError, ZErrorKind, ZResult};
+use zenoh_util::{zasynclock, zasyncread, zasyncwrite, zerror};
+
+// Default MTU (UNIX PDU) in bytes.
+// NOTE: Since this Unix Domain Socket is a byte-stream oriented transport,
+//       theoretically it has no limit regarding the MTU.
+//       However, given the batching strategy adopted in Zenoh
+//       and the usage of 16 bits in Zenoh to encode the
+//       payload length in byte-streamed, the UNIX MTU is constrained to
+//       2^16-1 bytes (i.e., 65535).
+const UNIX_MAX_MTU: usize = 65_535;
+
+zconfigurable! {
+    // Default MTU (UNIX PDU) in bytes.
+    static ref UNIX_DEFAULT_MTU: usize = UNIX_MAX_MTU;
+    // Size of buffer used to read from socket.
+    static ref UNIX_READ_BUFFER_SIZE: usize = 2*UNIX_MAX_MTU;
+    // Size of the vector used to deserialize the messages.
+    static ref UNIX_READ_MESSAGES_VEC_SIZE: usize = 32;
+    // Amount of time in microseconds to throttle the accept loop upon an error.
+    // Default set to 100 ms.
+    static ref UNIX_ACCEPT_THROTTLE_TIME: u64 = 100_000;
+}
+
+#[macro_export]
+macro_rules! get_unix_path {
+    ($locator:expr) => {
+        match $locator {
+            Locator::Unix(path) => path,
+            _ => {
+                let e = format!("Not a UNIX locator: {:?}", $locator);
+                log::debug!("{}", e);
+                return zerror!(ZErrorKind::InvalidLocator { descr: e });
+            }
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! get_unix_path_as_string {
+    ($locator:expr) => {
+        match $locator {
+            Locator::Unix(path) => match path.to_str() {
+                Some(path_str) => path_str.to_string(),
+                None => {
+                    let e = format!("Not a UNIX locator: {:?}", $locator);
+                    log::debug!("{}", e);
+                    "None".to_string()
+                }
+            },
+            _ => {
+                let e = format!("Not a UNIX locator: {:?}", $locator);
+                log::debug!("{}", e);
+                "None".to_string()
+            }
+        }
+    };
+}
+
+/*************************************/
+/*              LINK                 */
+/*************************************/
+pub struct Unix {
```

Comment:
We should avoid ambiguity on the usage of term Unix here. I suggest either to use:
- UnixDomainSocket
- Uds